### PR TITLE
ConstantsHelper::is_use_of_global_constant(): add tests

### PR DESCRIPTION
**Pull request by rodrigoprimo**
# Description

In preparation for supporting PHPCS 4.0, this PR adds unit tests for the `ConstantsHelper::is_use_of_global_constant()` method.

It also removes the now-redundant `@covers` tag for this method from `EscapeOutputUnitTest.php`, as the method is now covered by its own dedicated test.

## Suggested changelog entry

_N/A_

## Related issues/external references

Related to: #2272

### CODE REVIEW

File: WordPress/Tests/Helpers/ConstantsHelper/IsUseOfGlobalConstantUnitTest.inc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+use const SomeNamespace\{/* testUseConstStatementGrouped */ PHP_OS, PHP_VERSION_ID};
+
+use Foo\{Bar, /* testUseStatementGrouped */ PHP_OS};
+
+/*
+ * The below should be recognized as use of a global constant.
+ */
+
+/* testEchoStatement */
+echo PHP_OS;
```
Comment (dingo-d) [inline WordPress/Tests/Helpers/ConstantsHelper/IsUseOfGlobalConstantUnitTest.inc:120]: Should we add `(function() { echo FOO; })();` as a test, or is this redundant?

Comment (rodrigoprimo) [reply WordPress/Tests/Helpers/ConstantsHelper/IsUseOfGlobalConstantUnitTest.inc:120]: Thanks for reviewing this PR, @dingo-d. As far as I can see, such a test would not exercise any part of the code that is not already covered by the existing tests. The enclosing scope doesn't change the behavior of the tested method.

That said, is there anything in particular that you are concerned about that is currently not covered by the existing tests, or am I missing the reason that would justify this test that you are suggesting?

Comment (dingo-d) [reply WordPress/Tests/Helpers/ConstantsHelper/IsUseOfGlobalConstantUnitTest.inc:120]: Nah, was just looking at the test cases and wanted to see if I can come with some additional odd possible edge cases 😄
